src/domainobjects/generatable.py

```diff
@@ -39,90 +186,391 @@ def generate_random_string(self, length,
+        String
+            Combined "ticker.exchange_code"
+        """
+
         return '{0}.{1}'.format(ticker, exchange_code)
-    
+
     def generate_isin(self, coi, cusip):
-        return ''.join([coi, cusip, '4'])
-    
+        """ Appends two input values to "'coi''cusip''4'"
+
+        Parameters
+        ----------
+        coi : String
+            a country of issuance code
+        cusip : int
+            a CUSIP value
+
+        Returns
+        -------
+        String
+
+        """
+        return ''.join([coi, str(cusip), '4'])
+
     def generate_credit_debit(self):
+        """ Generate a random credit or debit value
+
+        Returns
+        -------
+        String
+            Random value between credit or debit
+        """
+
         return random.choice(self.CREDIT_DEBIT)
-    
+
     def generate_long_short(self):
+        """ Generate a random long or short value
+
+        Returns
+        -------
+        String
+            Random value between long or short
+        """
+
         return random.choice(self.LONG_SHORT)
-    
+
     def generate_position_type(self, no_sd=False, no_td=False):
+        """ Generate a random position type
+
+        Parameters
+        ----------
+        no_sd : Boolean
+            Boolean flag of whether to include settlement date in choice
+        no_td : Boolean
+            Boolean flad of whether to include trade date in choice
+
+        Returns
+        -------
+        String
+            Random value between of remaining set of possible position types
+        """
+
         choices = self.POSITION_TYPES
         if no_sd:
             choices.remove('SD')
         if no_td:
             choices.remove('TD')
         return random.choice(choices)
-    
+
     def generate_knowledge_date(self):
+        """ Generate a knowledge day value
+
+        Returns
+        -------
+        Date
+            Todays date
+        """
+
         return datetime.today()
-    
-    def generate_effective_date(self, n_days_to_add=3, knowledge_date=None, position_type=None):
-        return knowledge_date if position_type == 'SD' else knowledge_date + timedelta(days=n_days_to_add)
+
+    def generate_effective_date(self, n_days_to_add=3,
+                                knowledge_date=None, position_type=None):
+        """ Generates an Effective Date value
+
+        Parameters
+        ----------
+        n_days_to_add : int
+            Number of days to add to knowledge date before event takes effect
+        knowledge_date : Date
+            The knowledge date of the object being generated
+        position_type : String
+            The position type of the object being generated
+
+        Returns
+        -------
+        Date
+            Future date if input date was trade date, current date if input
+            date was settlement.
+        """
+
+        return knowledge_date if position_type == 'SD' \
+            else knowledge_date + timedelta(days=n_days_to_add)
```

Comment:
Do we have any weekend logic written up? We should add logic to skip weekend. (lets ignore holidays for now)